File: oa-portal/problems/m73-signal-reconstruction/starters/main.java
import java.io.*;
import java.util.*;

public class Main {

    /* Return the 1-based block {l, r} to reverse, or {-1, -1} if impossible. */
    static int[] findReversedBlock(long[] a) {
        // write your code here

        return new int[] { -1, -1 };
    }

    public static void main(String[] args) throws IOException {
        FastReader in = new FastReader();
        int n = in.nextInt();
        long[] a = new long[n];
        for (int i = 0; i < n; i++) a[i] = in.nextLong();

        int[] r = findReversedBlock(a);
        System.out.println(r[0] < 0 ? "-1" : (r[0] + " " + r[1]));
    }

    static class FastReader {
        private final DataInputStream in = new DataInputStream(new BufferedInputStream(System.in, 1 << 16));
        int nextInt() throws IOException { return (int) nextLong(); }
        long nextLong() throws IOException {
            int b = in.read();
            while (b != '-' && (b < '0' || b > '9')) b = in.read();
            boolean neg = b == '-';
            if (neg) b = in.read();
            long v = 0;
            while (b >= '0' && b <= '9') { v = v * 10 + (b - '0'); b = in.read(); }
            return neg ? -v : v;
        }
        String next() throws IOException {
            int b = in.read();
            while (b <= ' ') b = in.read();
            StringBuilder sb = new StringBuilder();
            while (b > ' ') { sb.append((char) b); b = in.read(); }
            return sb.toString();
        }
    }
}

File: oa-portal/problems/m73-signal-reconstruction/starters/main.cpp
#include <bits/stdc++.h>
using namespace std;

/* Return the 1-based block {l, r} to reverse, or {-1, -1} if impossible. */
pair<int,int> findReversedBlock(const vector<long long>& a) {
    // write your code here

    return make_pair(-1, -1);
}

int main() {
    int n;
    if (scanf("%d", &n) != 1) return 0;
    vector<long long> a(n);
    for (auto &x : a) scanf("%lld", &x);

    pair<int,int> r = findReversedBlock(a);
    if (r.first < 0) printf("-1\n");
    else printf("%d %d\n", r.first, r.second);
    return 0;
}
